src/bluesearch/database/article.py

```diff
@@ -202,44 +209,51 @@ def paragraphs(self) -> Generator[tuple[str, str], None, None]:
             The paragraph content.
         """
         # Abstract are removed because already parsed in abstract property.
-        abstract_p = self.content.findall(".//abstract/p")
+        abstract_p = self.content.findall("./front/article-meta/abstract//p")
         # Caption are removed because already parsed later in the current method
         # with the section name "Caption".
-        caption_p = self.content.findall(".//caption/p")
+        caption_p = self.content.findall("./body//caption/p")
         # Acknowledgements are removed because not useful.
-        acknowledg_p = self.content.findall(".//ack/p")
+        acknowledg_p = self.content.findall("./back/ack//p")
 
         exclude_list = abstract_p + caption_p + acknowledg_p
 
         # Paragraphs of text body
+        children_p = []
         section_dirs = self.get_paragraphs_sections_mapping()
         for paragraph in self.content.findall(".//p"):
-            if paragraph not in exclude_list:
+            # Check that paragraph is not in the exclude list and
+            # that paragraph is not child of another paragraph.
+            if paragraph not in exclude_list and paragraph not in children_p:
+                children_p += paragraph.findall(".//p")
                 text = self._element_to_str(paragraph)
                 section_title = ""
                 if paragraph in section_dirs:
                     section_title = section_dirs[paragraph]
-                yield section_title, text
+                if text:
+                    yield section_title, text
```

Comment:
Similarly to above, I'm still not 100% happy with the way we handle this. Like PA noted, keeping exclusion lists while doing global globbing looks rather hacky and I can't tell with certainty there won't be any edge cases that break this.

I might not have the full picture here, but by looking at one XML file I picked I see the following structure:
```xml
<body>
    <sec>
        <title>Section 1</title>
        <p>...</p>
        <p>...</p>
    </sec>
    <sec>
        <title>Section 1</title>
        <p>...</p>
        <p>...</p>
    </sec>
</body>
```
So what speaks against doing something like this:
```python
for sec in self.content.findall("./body"):
    sec_title = self._element_to_string(sec.find("title"))
    for par in sec.findall("p"):
        par_text = self._element_to_string(par)
        yield sec_title, par_text
```
plus adjusting `_element_to_string` so that it's a no-op for figures and tables because we want to extract them separately?

Reply:
I totally agree this comment. 

Maybe some comments: 
- Unfortunately, the structure is not always as straightforward as the one shown in your example: Paragraphs might not be contained in a section, there can have subsections, ... Moreover, from my experience, it is not possible to have a simple `XPath` catching all the paragraphs without any double slash (as illustrated with the previous example). 
- Just to clarify the purpose of the implementation: it seems sometimes paragraphs are children of paragraphs. To avoid any duplication of text, I decided to go for a solution parsing the parent paragraphs and avoid parsing again the children ones. 
- The exclude list logic was not introduced by this PR.

Reply:
Do we have examples of paragarphs in `<body>` that are not inside a section? I'd be curious to see that.

Apart from that, I would say that it shouldn't matter if there are subsections or paragraphs within paragraphs, or any other nested structure as this should all be already succcessfully handled by `_element_to_str`.

My general feeling is that JATS being a fixed schema, its structure should be 100% predictable and there shouldn't be any stray elements in places where we wouldn't expect them to be. That's the reason why I keep thinking it should be possible to do the parsing without globbing over the whole XML file.

Reply:
@Stannislav What do you think about the new implementation ?

Reply:
Nice, I love it :)) Thanks for implementing. I didn't test though, just read the code.

If you want to avoid the complexity with varying return types and the corresponding overloads one could just always return tuples `(title, text)`. Then for the abstracts one would probably do something like this:
```python
for _, text in self.parse_section(abstract):
    yield text
```
I think I'm fine either way. Up to you/others to decide :)

Reply:
Another small trick would be to avoid the checks `if abstract` and `if text` be letting `parse_section` accept `Element | None` and then check for `None` inside `parse_section`:
```python
def parse_section(element: Element | None):
    if element is None:
        return
```

Reply:
Thanks for the suggestions!
✅ Done the first one in d71e62b.